operate/qa/integration-tests/src/test/java/io/camunda/operate/opensearch/OpensearchFinishedImportingIT.java

```diff
@@ -0,0 +1,254 @@
+    EXPORTER.configure(exporterTestContext);
+    EXPORTER.open(new ExporterTestController());
+
+    recordsReaderHolder.resetCountEmptyBatches();
+  }
+
+  @Test
+  public void shouldMarkRecordReadersAsCompletedIf870RecordReceived() throws IOException {
+    // given
+    final var record = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 1);
+    EXPORTER.export(record);
+    osClient.index().refresh("*");
+
+    zeebeImporter.performOneRoundOfImport();
+
+    // when
+    final var record2 = generateRecord(ValueType.PROCESS_INSTANCE, "8.7.0", 1);
+    EXPORTER.export(record2);
+    osClient.index().refresh("*");
+
+    // receives 8.7 record and marks partition as finished importing
+    zeebeImporter.performOneRoundOfImport();
+
+    // then
+    // require multiple checks to avoid race condition. If records are written to zeebe indices and
+    // before a refresh, the record reader pulls the import batch is empty so it then says that the
+    // record reader is done when it is not.
+    for (int i = 0; i < RecordsReaderHolder.MINIMUM_EMPTY_BATCHES_FOR_COMPLETED_READER; i++) {
+      zeebeImporter.performOneRoundOfImport();
+    }
+
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(30))
+        .until(() -> isRecordReaderIsCompleted("1-process-instance"));
+  }
+
+  @Test
+  public void shouldMarkMultiplePositionIndexAsCompletedIf870RecordReceived() throws IOException {
+    final var processInstanceRecord = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 1);
+    final var decisionEvalRecord = generateRecord(ValueType.DECISION_EVALUATION, "8.6.0", 1);
+    final var decisionRecord = generateRecord(ValueType.DECISION, "8.6.0", 1);
+    EXPORTER.export(processInstanceRecord);
+    EXPORTER.export(decisionEvalRecord);
+    EXPORTER.export(decisionRecord);
+
+    osClient.index().refresh("*");
+    zeebeImporter.performOneRoundOfImport();
+
+    // when
+    final var newVersionProcessInstanceRecord =
+        generateRecord(ValueType.PROCESS_INSTANCE, "8.7.0", 1);
+    EXPORTER.export(newVersionProcessInstanceRecord);
+    final var newVersionDecisionEvalRecord =
+        generateRecord(ValueType.DECISION_EVALUATION, "8.7.0", 1);
+    EXPORTER.export(newVersionDecisionEvalRecord);
+
+    for (int i = 0; i <= RecordsReaderHolder.MINIMUM_EMPTY_BATCHES_FOR_COMPLETED_READER; i++) {
+      // simulate existing decision records left to process so it is not marked as completed
+      final var decisionRecord2 = generateRecord(ValueType.DECISION, "8.6.0", 1);
+      EXPORTER.export(decisionRecord2);
+      osClient.index().refresh("*");
+
+      zeebeImporter.performOneRoundOfImport();
+    }
+
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(30))
+        .until(
+            () ->
+                isRecordReaderIsCompleted("1-process-instance")
+                    && isRecordReaderIsCompleted("1-decision-evaluation")
+                    && !isRecordReaderIsCompleted("1-decision"));
```

Comment:
🤔 💭 This is a problem if we never mark our partition as fully complete

Reply:
Or would it be complete if we now run into 5 empty batches? ❓

Reply:
I'm not quite sure what you mean, on line 133 we run `MINIMUM_EMPTY_BATCHES_FOR_COMPLETED_READER` number of rounds of importing, and for the process instance and decision evaluation record readers this will result in 5 empty batches which given the exported 8.7 record (which marked the partition completed)